nntrainer/layers/layer_context.cpp

```diff
@@ -193,6 +203,19 @@ Tensor &RunLayerContext::getTensorGrad(unsigned int idx) {
   return tensors[idx]->getGradientRef();
 }
 
+/**
+ * @brief Get the Tensor Grad object
+ *
+ * @param idx Identifier of the tensor
+ * @return Tensor& Reference to the tensor grad tensor
+ */
+const Tensor &RunLayerContext::getTensorGrad(unsigned int idx) const {
+  if (!tensors[idx]->hasGradient())
```

Comment:
There are two checks:
1. was this tensor even requested - that will be done with #1458 using `.at()`
2. was a gradient requested for this tensor (was the request for trainable tensor or regular)

The check here performs the second check. So, this behavior cannot be employed with getTensor() as it only requires the first check.

Reply:
Yes, I mean https://github.com/nnstreamer/nntrainer/pull/1522#discussion_r717391889 for the getTensors() those checks aren't done.

Call to this function when !grad->allocated() will throw. But call to getTensor() with tensor->allocated() will not throw.

If they don't have a same semantics, it would cause some bugs later.

Reply:
But this is minor issue, and slightly off the topic of this PR. I'll approve

Reply:
Assuring that the tensor is allocated is part of the design itself. For all the tensors given out by the RunLayerContext, the assumption is that graph.allocate() will set and allocate them. So, the checks are not required by design.